src/run.cpp

```diff
@@ -153,7 +159,24 @@ kj::Promise<RunState> Run::start(RunState lastResult, std::shared_ptr<Context> c
         // main() by calling leader_main()
         char* procName;
         if(asprintf(&procName, "{laminar} %s:%d", name.data(), build) > 0)
+#ifndef __FreeBSD__
             execl("/proc/self/exe", procName, NULL); // does not return
+#else
+        {
+            char currproc_file[] = "/proc/curproc/file";
+            char *self_exe = currproc_file;
+            int  sysctl_rq[] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+            size_t sysctl_path_len = PATH_MAX;
+            char sysctl_path[PATH_MAX];
+
+            /* Sysctl is preferred way to get path of this program, if it fails try
+             * a fallback to '/proc/currproc/file', which is unlikely mounted by default */
+            if (!sysctl(sysctl_rq, 4, sysctl_path, &sysctl_path_len, NULL, 0))
+                self_exe = sysctl_path;
+
+            execl(self_exe, procName, NULL); // does not return
```

Comment:
I think it would be good to pick one way, either demand `proc` is mounted and fail noisily if not, or not require or fall back to it at all.

Reply:
ok